portfolio/src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.AnonymousUser;
+import com.google.sps.data.AuthenticatedUser;
+import com.google.sps.data.User;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles user accounts */
+@WebServlet("/user")
+public class UserServlet extends HttpServlet {
+
+  /**
+   * Handles the GET requests to "/user" path.
+   * Returns a JSON object with the user information.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    User user;
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (userService.isUserLoggedIn()) {
+      String userId = userService.getCurrentUser().getUserId();
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutURL = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      // Load user data from the database
+      Entity entity = getUser(userId);
+      // If the user data has not been stored in the database, then create the corresponding entity
+      if (entity == null) {
+        entity = new Entity("User", userId);
+        entity.setProperty("id", userId);
+        entity.setProperty("email", userEmail);
+        // By default, set the username to the email address without the domain 
+        entity.setProperty("username", (userEmail.split("@")[0]));
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
optional: it may be better if these could be moved to a method.

Reply:
Indeed! Done!